src/main/java/com/vg/hm/samples/usermanagementsvc/util/VehicleAPIRetrofitClient.java

```diff
@@ -0,0 +1,30 @@
+package com.vg.hm.samples.usermanagementsvc.util;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.vg.hm.samples.usermanagementsvc.service.VehicleAPIService;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+@Configuration
+public class VehicleAPIRetrofitClient {
+
+    @Bean(name = "vehicleServiceRetrofitClient")
+    public VehicleAPIService getRetrofit() {
+        String BASE_URL = "https://vpic.nhtsa.dot.gov/api/vehicles/";
```

Comment:
create a spring environment variable called "vehicle.service.base.url" on your application.properties file, and inject this value here. That way you don't hard code this value here. In the future we can switch the urls by updating the property file. 
Figure out how you can inject the a spring variable into a class.